spec/app/domain/util/open_ssl/x509/request_spec.rb

```diff
@@ -0,0 +1,185 @@
+__END__
+require 'spec_helper'
+
+private_key = OpenSSL::PKey::RSA.new(1048)
+altnames = ['URI:spiffe://cluster.local/namespace/foo/pod/bar']
+
+csr = CSR.new(
+  country: 'US',
+  state: 'CA',
+  city: 'San Francisco',
+  department: 'Web',
+  organization: 'Example Inc.',
+  common_name: 'example.com',
+  email: 'john@example.com',
+  private_key: private_key
+).request
+
+# prepare SAN extension
+extensions = [
+  OpenSSL::X509::ExtensionFactory.new.create_extension('subjectAltName', altnames.join(','))
+]
+
+# add SAN extension to the CSR
+attribute_values = OpenSSL::ASN1::Set [OpenSSL::ASN1::Sequence(extensions)]
+[
+  OpenSSL::X509::Attribute.new('extReq', attribute_values),
+  OpenSSL::X509::Attribute.new('msExtReq', attribute_values)
+].each do |attribute|
+  csr.add_attribute(attribute)
+end
+
+# sign CSR with the signing key
+csr.sign(private_key, OpenSSL::Digest::SHA256.new)
```

Comment:
Is this `csr` object being used anywhere? I don't see it referenced below.

Reply:
see above, this is all dead code now.